Assets/Scripts/Utility/MacroHelper.cs

```diff
@@ -541,8 +541,15 @@ public static string CityName(IMacroContextProvider mcp)
         }
 
         private static string CityName2(IMacroContextProvider mcp)
-        {   // %cn2 (only used in msg #200 where cn and cn2 are random? places)
-            throw new NotImplementedException();
+        {   // %cn2 (only used in msg #200 where cn2 is a random? city)
+            // Return first city of the region, unless it's current location, not worth doing more as the macro is hardly ever used.
+            DFRegion dfRegion = GameManager.Instance.PlayerGPS.CurrentRegion;
+            for (int i = 0; i < dfRegion.LocationCount; i++)
+            {
+                if (GameManager.Instance.PlayerGPS.CurrentLocationIndex != i && dfRegion.MapTable[i].LocationType == DFRegion.LocationTypes.TownCity)
+                    return dfRegion.MapNames[i];
+            }
+            return "Daggerfall";
```

Comment:
Thanks for fix. :) Ideal to use GetLocalizedText here. Regions are in database with region name as key. I'll patch this one after merge.